src/async_impl/client.rs

```diff
@@ -435,23 +432,22 @@ impl ClientBuilder {
         self
     }
 
-    /// Enable a persistent cookie store for the client.
+    /// Set the persistent cookie store for the client.
     ///
-    /// Cookies received in responses will be preserved and included in
-    /// additional requests.
+    /// Cookies received in responses will be passed to this store, and
+    /// additional requests will query this store for cookies.
     ///
     /// By default, no cookie store is used.
     ///
     /// # Optional
     ///
     /// This requires the optional `cookies` feature to be enabled.
     #[cfg(feature = "cookies")]
-    pub fn cookie_store(mut self, enable: bool) -> ClientBuilder {
```

Comment:
I completely forgot that this was actually proposing a breaking change. Sigh... Well, for 0.11.x, I guess we can live with `cookie_store(bool)` and `cookie_provider(dyn CookieStore)` (naming hard).